src/system/dev/vfs.c

```diff
@@ -155,6 +162,54 @@ int _close(int file) {
 	return ret;
 }
 
+int _mkdir(const char* path) {
+	struct _reent* r = _REENT;
+	if (!file_size(file)) {
+		r->_errno = ENAMETOOLONG;
+		return -1;
+	}
+	return file_table[file].driver->mkdir(r, file_table[file].arg, path);
+}
+
+int _unlink(char* name) {
+	struct _reent* r = _REENT;
+	return file_table[file].driver->mkdir(r, file_table[file].arg, path);
+}
+
+int _link(char* old, char* new) {
+	struct _reent* r = _REENT;
+	if (!file_size(file)) {
+		r->_errno = ENAMETOOLONG;
+		return -1;
+	}
+	return file_table[file].driver->link(r, file_table[file].arg, new, old);
+}
+
+int _stat(const char* restrict path, struct stat* st) {
+	struct _reent* r = _REENT;
+	return file_table[file].driver->stat(r, file_table[file].arg, path, st);
+}
+
+int _chdir(const char* path) {
+	struct _reent* r = _REENT;
+	return file_table[file].driver->chdir(r, file_table[file].arg, path);
+}
+
+int _chmod(const char* path, mode_t mode) {
+	struct _reent* r = _REENT;
+	return file_table[file].driver->chdir(r, file_table[file].arg, path, mode);
+}
+
+int _pathconf(const char* path, int name) {
+	struct _reent* r = _REENT;
+	return file_table[file].driver->pathconf(r, file_table[file].arg, path, name);
+}
+
+int _getcwd(char* buf, size_t size) {
+	struct _reent* r = _REENT;
+	return file_table[file].driver->getcwd(r, file_table[file].arg, buf, size);
```

Comment:
a year later, i'm not really sure what i was thinking... off the top of my head i guess the only real use for the concept of a current working directory is to resolve relative paths. additionally, it would _have_ to sit above the filesystem driver level; that's the only way it would make sense functionally. i'm leaning toward "just don't have a CWD" at this point, because none of the built-in filesystems support directories (either semantically or functionally), but i could be convinced otherwise

Reply:
I'm also not sure of the use-case, since you're only writing one "program". If what's needed is a clean way to pass in a working directory to a part of the program (which having a traditional working directory wouldn't solve due to it being global state), the [C++17 `std::filesystem::path` class](https://en.cppreference.com/w/cpp/filesystem/path) is probably the best route.

Reply:
that's a good point, though if we ever wanted to provide some kind of shell it might be useful

Reply:
either way it looks like the interim solution is to just mark directory functions as ENOSYS